nighthawk/BUILD

```diff
@@ -1,10 +1,24 @@
 load(
     "@envoy//bazel:envoy_build_system.bzl",
     "envoy_cc_binary",
+    "envoy_cc_library",
 )
 
 package(default_visibility = ["//visibility:public"])
 
+envoy_cc_library(
+    name = "hdrhistogram_c",
+    srcs = glob(["hdrhistogram_c/src/*.c"]),
+    hdrs = glob(["hdrhistogram_c/src/*.h"]),
+    copts = [
+        "-std=gnu99",
+        "-Wno-implicit-function-declaration",
+        "-Wno-error",
+    ],
+    repository = "@envoy",
```

Comment:
Shouldn't this be the hdrhistogram submodule repo? Or is it not a local repository and is just being overlaid on top of the main Envoy repo?

Reply:
It is being overlaid on top of the Envoy repo, without doing that I get `no such package 'bazel': BUILD file not found on package path and referenced by '//nighth
awk:hdrhistogram_c`

Reply:
Yeah, TBH this is a pretty weird way to manage external repositories. I think for HDR histogram, since we don't need the regular ability to instrument and hack that tree, it should not be modeled as a submodule. Instead, I think we should make this an external repository in the Bazel sense and inject a BUILD file into it.

For reference, here's the way we've done this in the main Envoy repository in an analogous situation https://github.com/envoyproxy/envoy/blob/84d038f5a67a8dc8ca9753e9089a33d8110db3c2/api/bazel/repositories.bzl#L34 (and other examples in that file). Basically, make it a `http_archive` and set `build_file_content`. Then you can reference it by `@whatever_name_you_set` in the `name` field.

Reply:
Do you want to sort this one out in this PR or in a followup PR?

Reply:
Moved this to the track-for-future list, I prefer doing it in a follow up.